otter/util/logging_treq.py

```diff
@@ -24,16 +24,23 @@ def _log_request(treq_call, url, **kwargs):
     """
     clock = kwargs.pop('clock', reactor)
     log = kwargs.pop('log', None)
+
     if not log:
         log = default_log
     method = kwargs.get('method', treq_call.__name__)
 
-    treq_transaction = str(uuid4())
+    kwargs.setdefault('headers', {})
+    treq_transaction = kwargs['headers'].setdefault('x-otter-request-id',
+                                                    [str(uuid4())])[0]
+
     log = log.bind(system='treq.request', url=url, method=method,
                    url_params=kwargs.get('params'),
                    treq_request_id=treq_transaction)
     start_time = clock.seconds()
 
+    kwargs.setdefault('headers', {})
+    kwargs['headers'].setdefault('x-otter-request-id', [treq_transaction])
```

Comment:
This is already set above right? Why is this set again? Sorry, I might be missing something here.